src/nspawn/nspawn-bind-user.c

```diff
@@ -279,11 +279,11 @@ int bind_user_prepare(
                 if (!GREEDY_REALLOC(c->data, c->n_data + 1))
                         return log_oom();
 
-                sm = strdup(u->home_directory);
+                sm = strdup(user_record_home_directory(u));
```

Comment:
@poettering This falls back to `/` if `homeDirectory` is not set so we bind mount the rootfs of the host into the container then? That doesn't seem right. Shouldn't we fail if the user has no home directory in its record or skip the mount?